sdks/go/pkg/beam/core/runtime/exec/pcollection.go

```diff
@@ -85,8 +86,8 @@ func (w *byteCounter) Write(p []byte) (n int, err error) {
 
 // ProcessElement increments the element count and sometimes takes size samples of the elements.
 func (p *PCollection) ProcessElement(ctx context.Context, elm *FullValue, values ...ReStream) error {
-	cur := atomic.AddInt64(&p.elementCount, 1)
-	if cur == p.nextSampleIdx {
+	cur := atomic.AddInt64(&p.bundleElementCount, 1)
+	if cur+p.pCollectionElementCount == p.nextSampleIdx {
```

Comment:
Consider adding a comment here and/or where p.pCollectionElementCount is defined about the goal of this variable, and why it's safe to do this (local use only, instances aren't shared etc). I'm pretty confident in this change as is, but lets make it obvious to the next reader.

Reply:
Done: ) 
Added in pCollection struct definition.